### ISSUE TITLE
Custom NPCs compatibility

### ISSUE THREAD

**Issue by FebKB** [Jackiecrazy/WarDance#59]
Sorry for my last post and not replying in months, I completely forgot about it, but here goes.
I love this mod and I would like to use it in my modpack "Chivalry and Wizardry". It works wonders except for one thing.
It completely screws over two main aspects of CustomNPCs.
1. Because of stealth mechanics added by the mod, it overrides the one present for CustomNPCs. It's great that they need direct line of sight, but it also limits their view. If I want an archer to detect you from 32 blocks away, it doesn't work because the range is overriden and it limits them.
2. For some reason, the health is also completely jumbled. When an NPC tries to regenerate back to Full HP, they are usually left with 1 less health point and, if I try to save them with the Cloner, their health values are decreased (ie. If it was 24 it turns to 23.33)
The version is 1.16.5, latest version for both. I don't know if there's a way to disable mechanics for certain NPCs but, if there are, please let me know.

-Once again sorry for not answering at all in my previous post, it's my bad-

**Comment by Jackiecrazy** [comment]
1. stealth mechanics are now in cloak and dagger. You can also blacklist custom NPCs with the config there. This is true of the latest version of 1.16.5.
2. Wounding may be to blame. It wasn't a very well thought-out mechanic... you can disable it in the config, though I can't remember which off the top of my head.

**Comment by FebKB** [comment]
Thank you, I will try it out ans report back if anything changes

**Comment by FEBtheKB** [comment]
So, I have not worked on the stealth mechanics, but Cloak & Dagger seems promising since it overrides Wardance and works in complete conjunction with it. Because of it, I think I'll be able to solve the first issue (still gotta test it when I have time), but for the second one, I've dug all I could and I did not find anything related to wounding or blacklisting mobs.

**Comment by Jackiecrazy** [comment]
Cloak&Dagger doesn't override anything from PWD. PWD used to include stealth mechanics, and they were stripped out into C&D by popular demand. Wounding, I recall, is in the combat or general config under "decay blacklist". I'm pretty sure there's also a wounding percentage option that you can just set to 0.

**Comment by FEBtheKB** [comment]
Hey, it seems to work! I tried configuring the NPCs as blacklisted on both stealth mechanics (achmnw) and for Wounding and it works wonders. Their health is no longer screwed when they regenerate and they seem to be doing much better. One thing I want to ask before going, I've seen that when you repeateadly hit an iron golem and then regenerate him to full hp with ingots, the next attack will do monstrous damage, even if you're unarmed.
Is this still a problem with wounding?

**Comment by Jackiecrazy** [comment]
...huh, that's an interesting issue. Are you letting its posture regenerate in between? It could be because the golem is staggered, or you have some skill equipped. Other than that, I don't think I'm the one doing it.

**Comment by FEBtheKB** [comment]
No, I am not letting its posture regenerate. I basically hit him with a bunch of attacks (I have an Apotheosis crossbow with Crescendo of Bolts) and after getting him nearly killed, I then proceed to heal him with ingots. When that happens, the golem will also take 10 damage out of nowhere, while healing, and if I land a succesful hit as soon as possibile, it will do an absurd amount of damage.

**Comment by Jackiecrazy** [comment]
I barely touch ranged combat, and 1.16 staggered does not deal much extra damage (it primarily negates armor), so I'm about 80% sure that's not me. Can you replicate this without PWD?

**Comment by FebKB** [comment]
I wasn't able to do it in these past few days, but I will see in the near future

**Comment by FebKB** [comment]
Ay, so I have finally tested it and I stand corrected.
I loaded a new world (still in the Chivalry and Wizardry modpack I made) and I took a high damaging sword (12 damage) and an iron golem as a test subject.
I regularly took him down to almost dying, then healed him back to full hp with ingots. Then I would punch, seeing what would happen.
When wounding was on, I would do upwards of 60 damage with a single punch. This is especially clear on repeated attempts.
When wounding was off, although it took some time in-game for it to load the config, damage was regular with any weapon, fists included.
Remember, this was done on a flat world with nothing on it. No skill were used for this test, judt a heavy hitting sword and golems.
This is important for CustomNPCs too, because wounding seems to just screw up health values for all entities affected.

**Comment by Jackiecrazy** [comment]
Interesting. My guess is that CustomNPCs or another mod is setting max health in another way, then when wounding inexplicably updates the mob is suddenly aware that it has too much health relative to its wounded max health and tries to compensate by damaging itself back to normal. In this case, it's fine to just set the wounding percentage to 0. I did away with wounding in 1.19+ for this reason.
